autoconnect/autoconnect-settings/src/app_state.rs

```diff
@@ -0,0 +1,80 @@
+use std::sync::Arc;
+
+use cadence::StatsdClient;
+use fernet::{Fernet, MultiFernet};
+
+use autoconnect_common::registry::ClientRegistry;
+use autopush_common::db::{client::DbClient, dynamodb::DdbClientImpl, DbSettings, StorageType};
+use autopush_common::{
+    errors::{ApcErrorKind, Result},
+    metrics::new_metrics,
+};
+
+use crate::{Settings, ENV_PREFIX};
+
+#[derive(Clone)]
+pub struct AppState {
+    /// Handle to the data storage object
+    pub db: Box<dyn DbClient>,
+    pub metrics: Arc<StatsdClient>,
+
+    /// Encryption object for the endpoint URL
+    pub fernet: MultiFernet,
+    /// The connected WebSocket clients
+    pub registry: Arc<ClientRegistry>,
+
+    pub settings: Arc<Settings>,
+    pub router_url: String,
+    pub endpoint_url: String,
+}
+
+impl AppState {
+    pub fn from_settings(settings: Settings) -> Result<Self> {
```

Comment:
I keep wondering if this should be a `TryFrom`

Reply:
Same -- I found [this comment](https://www.reddit.com/r/rust/comments/k7z66j/comment/gev84pr/?utm_source=share&utm_medium=web2x&context=3) with a pretty reasonable take on this, looking at the intent (conversion vs construction): "for instance I can construct a db connection from a string saying that I can convert a string to a db connection would be… odd". `AppState` and e.g. `StorageType::from_dsn` that it calls are definitely  construction.